domain/consensus/consensus.go

```diff
@@ -70,42 +79,69 @@ func validateTransactionInContextAndPopulateMassAndFeeVirtualBlockHash() *extern
 	panic("unimplemented")
 }
 
+func isHeaderInPruningPointFutureAndVirtualPast() bool {
+	panic("unimplemented")
+}
+
 func (s *consensus) GetBlock(blockHash *externalapi.DomainHash) (*externalapi.DomainBlock, error) {
-	panic("implement me")
+	return s.blockStore.Block(s.databaseContext, blockHash)
 }
 
 func (s *consensus) GetBlockHeader(blockHash *externalapi.DomainHash) (*externalapi.DomainBlockHeader, error) {
-	panic("implement me")
+	return s.blockHeaderStore.BlockHeader(s.databaseContext, blockHash)
 }
 
 func (s *consensus) GetBlockInfo(blockHash *externalapi.DomainHash) (*externalapi.BlockInfo, error) {
-	panic("implement me")
+	blockInfo := &externalapi.BlockInfo{}
+
+	exists, err := s.blockStatusStore.Exists(s.databaseContext, blockHash)
+	if err != nil {
+		return nil, err
+	}
+	blockInfo.Exists = exists
+	if !exists {
+		return blockInfo, nil
```

Comment:
We should probably return something that is not an empty BlockInfo to signal that block doesn't exist (I'd say return nil, but you'll shoot me for such insolent behavior)

Reply:
We do return blockInfo.Exists = false.